COMP371/Main.cpp

```diff
@@ -196,11 +204,30 @@ void render() {
 	chunkNormalsShader->setUniform("vpMatrix", RenderingContext::get()->camera.getViewProjectionMatrix());
 #endif
 
-	// Render chunks
+	const std::unordered_map<int64, Chunk>& chunks = ChunkManager::instance()->getCurrentlyLoadedChunks();
+	shadowMap->updateMvp(lightDirection);
+
+	//First Pass (Shadows)
+	glBindFramebuffer(GL_FRAMEBUFFER, shadowMap->getFbo());
+	smShader->use();
+	smShader->setUniform("lightSpaceMatrix", shadowMap->getMvp());
+	glClear(GL_DEPTH_BUFFER_BIT);
```

Comment:
Is it normal that the color buffer isn't being cleared?

Reply:
Yeah, color is not being written, only depth info. No need to reset the color bit. Also, we're waiting for the water shader PR to merge into master.

Reply:
Yeh I just need to make the hard coded values uniforms, but im busy with the assignment 3 and lab reports right now